src/analytics/SegmentAnalyticsService.js

```diff
@@ -155,8 +155,17 @@ class SegmentAnalyticsService {
    * @param {*} [traits]
    */
   identifyAnonymousUser(traits) {
-    global.analytics.identify(traits);
-    this.hasIdentifyBeenCalled = true;
+    // if we do not have an authenticated user (indicated by being in this method),
+    // but we still have a user id associated in segment, reset the local segment state
+    // This has to be wrapped in the analytics.ready call because the analytics.user()
+    // function isn't available until analytics is loaded.
+    global.analytics.ready(() => {
```

Comment:
So I think perhaps what we should do is make `identifyAnonymousUser` return a promise so we can `await` it in the `initialize` function.  That way we don't proceed with initialization without having finished identifying the user - assuming that `global.analytics.ready` is asynchronous.  Then in initialize, we can just call `await identifyAnonymousUser` and all will be good.